tests/unit/models/test_model_interfaces.py

```diff
@@ -357,6 +357,15 @@ def test_gaussian_process_regression_pairwise_covariance(gpr_interface_factory)
     np.testing.assert_allclose(expected_covariance, actual_covariance, atol=1e-5)
 
 
+def test_gpr_raises_for_invalid_init() -> None:
```

Comment:
we can be more specific here, using `num_prior_samples` instead of `init` in the title